src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -734,7 +737,8 @@ static int enumerate_partitions(dev_t devnum) {
         }
 
         if (m->partitions[PARTITION_ESP].found) {
-                k = add_partition_esp(m->partitions + PARTITION_ESP, m->partitions[PARTITION_XBOOTLDR].found);
+                k = add_partition_esp(m->partitions + PARTITION_ESP, m->partitions[PARTITION_XBOOTLDR].found,
+                                      m->partitions[PARTITION_ROOT].found);
```

Comment:
this is kinda pointless. when called after the initrd transition gpt-auto-generator already operates on the device anyway that backs the current root device.

moreover this check you added here actually checks for "hey, does this block device have a properly tagged root partition", which is quite different from "does this block device have the root partition we are currently using on it?", which your commit msg suggests you are looking for (but which is pointless, as mentioned).

Reply:
Oh, I misunderstood that API. Fair enough.